streams/src/main/java/org/apache/kafka/streams/query/RangeQuery.java

```diff
@@ -59,25 +58,25 @@ private RangeQuery(final Optional<K> lower, final Optional<K> upper, final boole
      * @param <V> The value type
      */
     public static <K, V> RangeQuery<K, V> withRange(final K lower, final K upper) {
-        return new RangeQuery<>(Optional.ofNullable(lower), Optional.ofNullable(upper), true);
+        return new RangeQuery<>(Optional.ofNullable(lower), Optional.ofNullable(upper), ResultOrder.ANY);
     }
 
     /**
-     * Determines if the serialized byte[] of the keys in ascending order.
+     * Determines if the serialized byte[] of the keys in ascending or any order.
      * Order is based on the serialized byte[] of the keys, not the 'logical' key order.
-     * @return true if ascending, false otherwise.
+     * @return return the order of returned records based on the serialized byte[] of the keys(can be unordered, or in ascending or in descending order).
      */
-    public boolean isKeyAscending() {
-        return isKeyAscending;
+    public ResultOrder resultOrder() {
+        return order;
     }
 
     /**
      * Set the query to return the serialized byte[] of the keys in descending order.
      * Order is based on the serialized byte[] of the keys, not the 'logical' key order.
-     * @return a new RangeQuery instance with descending flag set.
+     * @return a new RangeQuery instance with DESCENDING set.
      */
     public RangeQuery<K, V> withDescendingKeys() {
```

Comment:
We should also add `withAscendingKeys()` method

Reply:
ok